src/protocol.rs

```diff
@@ -292,52 +292,162 @@ impl SignatureShare {
+                );
+                Err(CallerError::ParticipantConfigError)?
+            }
+
+            // Test for uniqueness
+            let unique_set =
+                HashSet::<&ParticipantIdentifier>::from_iter(std::iter::once(&id).chain(other_ids));
+            if unique_set.len() != other_ids.len() + 1 {
+                error!(
+                    "Tried to create a participant config with a non-unique set of participants"
+                );
+                Err(CallerError::ParticipantConfigError)?
+            }
+
+            Ok(Self {
+                id,
+                other_ids: other_ids.to_vec(),
             })
-            .collect())
+        }
+
+        /// Get a list of `size` consistent [`ParticipantConfig`]s.
+        ///
+        /// Each config contains a different permutation of a single overall set
+        /// of [`ParticipantIdentifier`]s.
+        ///
+        /// **⚠️ Security warning:** This method implies the existence of a
+        /// trusted third party that generates the participant IDs. This
+        /// method must not be used if your deployment does not have a
+        /// trusted party.
+        pub fn random_quorum<R: RngCore + CryptoRng>(
+            size: usize,
+            rng: &mut R,
+        ) -> Result<Vec<ParticipantConfig>> {
+            if size < 2 {
+                Err(CallerError::ParticipantConfigError)?
+            }
+            let ids = std::iter::repeat_with(|| ParticipantIdentifier::random(rng))
+                .take(size)
+                .collect::<Vec<_>>();
+
+            (0..size)
+                .map(|i| {
+                    let mut other_ids = ids.clone();
+                    let id = other_ids.swap_remove(i);
+                    Self::new(id, other_ids.as_slice())
+                })
+                .collect::<Result<_>>()
+        }
+
+        ///Create a random [`ParticipantConfig`].
+        #[cfg(test)]
+        pub(crate) fn random<R: RngCore + CryptoRng>(
+            size: usize,
+            rng: &mut R,
+        ) -> ParticipantConfig {
+            assert!(size > 1);
+            let other_ids = std::iter::repeat_with(|| ParticipantIdentifier::random(rng))
+                .take(size - 1)
+                .collect::<Vec<_>>();
+            let id = ParticipantIdentifier::random(rng);
+            Self { id, other_ids }
+        }
     }
 
-    ///Create a random [`ParticipantConfig`].
     #[cfg(test)]
-    pub(crate) fn random<R: RngCore + CryptoRng>(size: usize, rng: &mut R) -> ParticipantConfig {
-        assert!(size > 1);
-        let other_ids = std::iter::repeat_with(|| ParticipantIdentifier::random(rng))
-            .take(size - 1)
-            .collect::<Vec<_>>();
-        let id = ParticipantIdentifier::random(rng);
-        Self { id, other_ids }
+    mod tests {
+        use super::*;
+        use crate::utils::testing::init_testing;
+
+        #[test]
+        fn participant_config_must_have_at_least_two_participants() {
```

Comment:
note: this test was modified from its original version; it used to push the badly-formed config all the way to `Participant` creation, but now that gets caught sooner.